runtime/vm/CRIUHelpers.cpp

```diff
@@ -1793,26 +1802,27 @@ criuCheckpointJVMImpl(JNIEnv *env,
 			vmFuncs->setLogOptions(vm, syslogOptions);
 		}
 		j9mem_free_memory(syslogOptions);
-		if (systemReturnCode < 0) {
-			currentExceptionClass = vm->checkpointState.criuSystemCheckpointExceptionClass;
-			nlsMsgFormat = j9nls_lookup_message(J9NLS_DO_NOT_PRINT_MESSAGE_TAG | J9NLS_DO_NOT_APPEND_NEWLINE, J9NLS_VM_CRIU_DUMP_FAILED, NULL);
-			goto wakeJavaThreadsWithExclusiveVMAccess;
-		}
 
-		/* We can only end up here if the CRIU restore was successful */
-		isAfterCheckpoint = TRUE;
+		if (criuDumpReturnCode >= 0) {
+			/* Set this if the dump succeeded. If it doesnt succeed we still need to run some of the restore
+			 * code as some threads are waiting to be notified.
+			 */
+			isAfterCheckpoint = TRUE;
+		}
 
-		switch (loadRestoreArguments(currentThread, optionsFileChars, envFileChars)) {
-		case RESTORE_ARGS_RETURN_OPTIONS_FILE_FAILED:
-			currentExceptionClass = vm->checkpointState.criuJVMRestoreExceptionClass;
-			nlsMsgFormat = j9nls_lookup_message(J9NLS_DO_NOT_PRINT_MESSAGE_TAG | J9NLS_DO_NOT_APPEND_NEWLINE,
-				J9NLS_VM_CRIU_LOADING_OPTIONS_FILE_FAILED, NULL);
-				/* fallthrough */
-		case RESTORE_ARGS_RETURN_OOM:
-			/* exception is already set */
-			goto wakeJavaThreadsWithExclusiveVMAccess;
-		case RESTORE_ARGS_RETURN_OK:
-			break;
+		if (isAfterCheckpoint) {
```

Comment:
Should the following `runInternalJVMRestoreHooks()` and `j9gc_reinitialize_for_restore()` proceed only when `isAfterCheckpoint` is `true`?

Reply:
no GC and JIT hooks need to run unconditionally so the JVM can coninue to execute normally if checkpoint fails